clang/lib/Sema/BoundsAnalysis.cpp

```diff
@@ -687,13 +687,11 @@ T BoundsAnalysis::Intersect(T &A, T &B) const {
     return Ret;
   }
 
-  for (auto I = Ret.begin(), E = Ret.end(); I != E;) {
+  for (auto I = Ret.begin(); I != Ret.end();) {
```

Comment:
LLVM coding guidelines recommend not to write iterator-based loops in this form, in general. This is because `Ret.end()` would be evaluated each time through the loop. However, since the container is mutated here we need to write the loop in this form.

Could you please add a comment here indicating that the reason we need to evaluate `Ret.end()` each time is because the container is mutated in the loop?

For more details see https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

Reply:
Done.